### DISCUSSION TITLE
CURL SSL certificate problem - Due to Company Decryption

### DISCUSSION THREAD

**Discussion by quella** [FreshRSS/FreshRSS#4345] [Q&A] [answered]
I'm running the Docker version of the FreshRSS via Docker Desktop which works wonderful and setup is quick.  Recently I did notice some CURL errors relating to the SSL Certificate Problems (cURL error 60: SSL certificate problem: self signed certificate in certificate chain [https://twistedsifter.com/feed/]) and I believe this is due to our company doing SSL Decryption (man-in-the-middle).  Wondering if there is an easy way to disable this certificate checking within the Docker Image or not for my instance. 

I was looking at this website:  https://fuchsia.googlesource.com/third_party/curl/+/curl-7_52_1/docs/SSLCERTS.md#:~:text=With%20libcurl%20you%20disable%20this,this%20with%20%2Dk%2F%2D%2Dinsecure.&text=Add%20the%20CA%20cert%20for,existing%20default%20CA%20certificate%20store.

But it was unclear on how I could implement this easily in the Docker image.  Ideas or direction?

**Comment by Alkarex** [comment, marked as answer]
Hello,
If you can get cURL to work, then it should be possible to get FreshRSS to work as well.
To tweak cURL options, some are exposed at feed level (e.g. disable certificate verification), and global options can be set in `data/config.php`, see example in
https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md#modify-the-configuration-of-a-running-freshrss-instance

https://github.com/FreshRSS/FreshRSS/blob/ae097f5bd82aa24ea19cca72ac9d1594b0dcaec0/config.default.php#L120-L131

See https://php.net/function.curl-setopt for a full list of options

**Reply by quella** [reply]
Alkarex thanks for the input, links, and ideas.  I will have a look within the docker container to where I could modify these settings.  Once there, I'm assuming I will need to ensure I modify any future releases as I update.  Love the tool and use it multiple time daily.

**Reply by Alkarex** [reply]
Your own `data/config.php` will not be modified during future updates, as long as you use a Docker volume as recommended, so the settings should persist.

**Reply by quella** [reply]
You are correct sir, as this data will remain in the same volume store.  Thanks again for the pointer.  That is even more helpful.

**Reply by quella** [reply]
Just a quick check before I send this config up to the docker container...  After applying, I can simply reboot the container to take effect.  My question is to ensure the syntax below is correct...  I have to use the ' character and have the spacing the same, correct?

  'curl_options' => 
  array (
   	# Options to disable SSL/TLS certificate check (e.g. for self-signed HTTPS) 
 	'CURLOPT_SSL_VERIFYHOST => 0, 
 	'CURLOPT_SSL_VERIFYPEER => false,
  ),

**Reply by Alkarex** [reply]
No need to restart the container, and no quote in the names:

```php
'curl_options' => array( 
 	# Options to disable SSL/TLS certificate check (e.g. for self-signed HTTPS) 
 	CURLOPT_SSL_VERIFYHOST => 0, 
 	CURLOPT_SSL_VERIFYPEER => false, 
 ), 
```

**Comment by quella** [comment]
Roger. Thank you sir for the confirmation. I’ll apply it sounded n the morning. Have a good evening.

**Comment by quella** [comment]
Worked like a charm and now I have no SSL errors in my logs.  Thanks for the pointers and direction.
